types/index.d.ts

```diff
@@ -1,4 +1,4 @@
-// Minimum TypeScript Version: 3.4
+// Minimum TypeScript Version: 3.8
```

Comment:
TS 3.4 is still supported.
This is bumped to 3.8 to allow adding `// $ExpectError` assertions/test cases on conflicting generic interfaces